toffy/streak_detection.py

```diff
@@ -0,0 +1,409 @@
+    """Uses the filtered streak dataframe to create a binary mask, where 1 indicates the pixels
+    that will get corrected. This mask can be later saved and used for visualization purposes.
+
+    Args:
+        streak_data (StreakData): An instance of the StreakData Dataclass, holds all necessary
+        data for streak correction.
+    """
+    streak_data.filtered_streak_mask = np.zeros(shape=streak_data.shape, dtype=np.int8)
+    for region in streak_data.filtered_streak_df.itertuples():
+        streak_data.filtered_streak_mask[
+            region.min_row : region.max_row, region.min_col : region.max_col
+        ] = 1
+
+
+def _make_box_outline(streak_data: StreakData) -> None:
+    """Creates a box outline for each binary streak using the filtered streak dataframe. Outlines
+    the streaks that will get corrected. This mask can be later saved and used for visualization
+    purposes.
+
+    Args:
+        streak_data (StreakData): An instance of the StreakData Dataclass, holds all necessary
+        data for streak correction.
+    """
+    streak_data.boxed_streaks = np.zeros(shape=streak_data.shape, dtype=np.int8)
+    for region in streak_data.filtered_streak_df.itertuples():
+        y, x = draw.rectangle_perimeter(
+            start=(region.min_row, region.min_col),
+            end=(region.max_row - 1, region.max_col - 1),
+            clip=True,
+            shape=streak_data.shape,
+        )
+        streak_data.boxed_streaks[y, x] = 1
+
+
+def _make_correction_mask(streak_data: StreakData) -> None:
+    """Creates the correction mask for each binary streak using the filtered streak DataFrame.
+    Marks pixels which will be used for the correction method.
+
+    Args:
+        streak_data (StreakData): An instance of the StreakData Dataclass, holds all necessary
+        data for streak correction.
+    """
+    padded_image = np.pad(
+        np.zeros(shape=streak_data.shape, dtype=np.int8), pad_width=(1, 1), mode="edge"
+    )
+
+    for region in streak_data.filtered_streak_df.itertuples():
+        padded_image[region.min_row, region.min_col + 1 : region.max_col + 1] = np.ones(
+            shape=(region.max_col - region.min_col)
+        )
+        padded_image[region.max_row + 1, region.min_col + 1 : region.max_col + 1] = np.ones(
+            shape=(region.max_col - region.min_col)
+        )
+
+    streak_data.corrected_streak_mask = util.crop(padded_image, crop_width=(1, 1))
+
+
+def _correct_streaks(streak_data: StreakData, input_image: np.ndarray) -> np.ndarray:
+    """Corrects the streaks for the input image. Uses masks in the streak_data Dataclass.
+    Performs the correction by averaging the pixels above and below the streak.
+
+    Args:
+        streak_data (StreakData): An instance of the StreakData Dataclass, holds all necessary
+        data for streak correction.
+        input_image (np.ndarray): The input image (Numpy array representation of the tiff file)
+        to be used for streak detection.
```

Comment:
This description of input image makes it seem like this is the noodle channel, not the channel getting corrected